sympy/core/add.py

```diff
@@ -90,9 +90,14 @@ def flatten(cls, seq):
 
         """
         from sympy.calculus.util import AccumBounds
+        from sympy import Poly
```

Comment:
Can you give a specific example? I can't reproduce this. For me something like `Poly(x, x) in [S.true, S.false]` gives AttributeError.